jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/wizard/SchemaSelectionPanel.kt

```diff
@@ -3,142 +3,92 @@
 
 package software.aws.toolkits.jetbrains.services.lambda.wizard
 
-import com.fasterxml.jackson.databind.JsonNode
-import com.intellij.openapi.project.Project
+import com.intellij.openapi.progress.ProgressIndicator
+import com.intellij.openapi.project.DefaultProjectFactory
+import com.intellij.openapi.roots.ModifiableRootModel
 import com.intellij.openapi.ui.ValidationInfo
+import com.intellij.openapi.vfs.VfsUtil
+import com.intellij.ui.layout.panel
 import software.amazon.awssdk.services.lambda.model.Runtime
-import software.aws.toolkits.jetbrains.core.credentials.ConnectionSettings
+import software.aws.toolkits.jetbrains.services.lambda.BuiltInRuntimeGroups
+import software.aws.toolkits.jetbrains.services.lambda.RuntimeGroup
 import software.aws.toolkits.jetbrains.services.lambda.runtimeGroup
-import software.aws.toolkits.jetbrains.services.schemas.SchemaDownloader
-import software.aws.toolkits.jetbrains.services.schemas.SchemaSummary
-import software.aws.toolkits.jetbrains.services.schemas.SchemaTemplateExtraContext
-import software.aws.toolkits.jetbrains.services.schemas.SchemaTemplateParameters
+import software.aws.toolkits.jetbrains.services.lambda.sam.SamCommon
+import software.aws.toolkits.jetbrains.services.lambda.sam.SamSchemaDownloadPostCreationAction
+import software.aws.toolkits.jetbrains.services.schemas.SchemaCodeLangs
+import software.aws.toolkits.jetbrains.ui.connection.AwsConnectionSettingsSelector
+import software.aws.toolkits.resources.message
 import javax.swing.JComponent
-import javax.swing.JLabel
-import javax.swing.JPanel
 
-// UI for selecting a Schema
-interface SchemaSelectionPanel {
-    val schemaSelectionPanel: JComponent
-
-    val schemaSelectionLabel: JLabel?
-
-    fun registryName(): String? = null
-
-    fun schemaName(): String? = null
-
-    fun reloadSchemas(awsConnection: ConnectionSettings? = null) {}
-
-    fun buildSchemaTemplateParameters(): SchemaTemplateParameters?
-
-    fun validateAll(): List<ValidationInfo>? = null
-
-    companion object {
-
-        @JvmStatic
-        fun create(
-            runtime: Runtime,
-            selectedTemplate: SamProjectTemplate,
-            generator: SamProjectGenerator
-        ): SchemaSelectionPanel =
-            runtime.runtimeGroup?.let { runtimeGroup ->
-                if (selectedTemplate.supportsDynamicSchemas())
-                    SamProjectWizard.getInstance(runtimeGroup).createSchemaSelectionPanel(generator)
-                else
-                    NoOpSchemaSelectionPanel()
-            } ?: NoOpSchemaSelectionPanel()
+/*
+ * A panel encapsulating  AWS credential selection during SAM new project creation wizard
+  */
+class SchemaSelectionPanel : WizardFragment {
+    private val schemaSelector = SchemaResourceSelector()
+    private val awsConnectionSelector = AwsConnectionSettingsSelector(
+        DefaultProjectFactory.getInstance().defaultProject,
+        schemaSelector::reloadSchemas
+    )
+    private val component = panel {
```

Comment:
dsl finally stable enough?

Reply:
Still playing with it, it seems to be, at lest from a source compat stand point